File: projects/04/factorial.rb
#! /usr/bin/ruby
# factorial.rb displays factorial of n
#
# Completed by: David Choi
#
# Input:  n, an integer
# Precondition: n >= 1
# Output: n!


if __FILE__ == $0
   print "To calculate n!, enter n: "
   n = gets.chomp.to_f
   answer = 1
   
   for i in 2..n
    answer = answer * i   
   end

   puts "The factorial of #{ n } is #{ answer }"
   puts "\n"
end

File: projects/04/Factorial.java
/* Factorial.java displays factorial of n.
 *
 * Input: An integer.
 * 
 * Output: A factorial value of N
 * 
 * Completed by: David Choi
 ********************************************************/
import java.util.Scanner; 		// Get input

class Factorial{

  public static void main(String[] args){

    Scanner keyboard = new Scanner(System.in);
    System.out.println("To calculate n!, enter n: ");
   
    double n = keyboard.nextDouble(); //input
    double b = 1; //answer
    
    for(int i = 2; i <= n ; i++)
    	b *= i;
    System.out.println("Factorial of " + n + " is " + b);
  }
}

